sites/website/src/docs/advanced/dependency-injection.md

```diff
@@ -5,7 +5,129 @@ sidebar_label: Dependency Injection
 custom_edit_url: https://github.com/microsoft/fast/edit/master/sites/website/src/docs/apps-and-experiences/dependency-injection.md
 description: FAST introduces the concept of a dependency injection container.
 keywords:
-    - dependency injection container
+    - dependency injection
+    - container
+    - createContainer
+    - getOrSetDOMContainer
 ---
 
-// TODO
\ No newline at end of file
+## Introduction
+
+Dependency injection is a flexible & modular design pattern for passing functionality to the parts of an app that need it. If you need your project to be more maintainable and testable, this might be a good solution for you to use. The `@microsoft/fast-element` package has dependency injection utilities for regular use in JavaScript, and for injecting dependencies into web components.
+
+### Basic Example
+
+The first step to dependency injection is the creation of a container, this is where dependencies will be injected and resolved. There are two available ways to create a container, you can either use `createContainer()` or `getOrCreateDOMContainer()`. You can have as many containers as you like, however they will only resolve dependencies registered within their container.
+
+```ts
+import { DI } from "@microsoft/fast-element/di.js";
+
+const container = DI.createContainer();
+```
+
+Now that your container is created, let's create a few dependencies to inject into it.
+
+First let's define an interface for the dependency which we'll call `MyServiceConfig`, and create a `ContextDecorator` of the same name which will take the interface as it's generic type:
+
+```ts
+import { DI } from "@microsoft/fast-element/di.js";
+import type { ContextDecorator } from "@microsoft/fast-element/context.js";
+
+export interface MyServiceConfig {
+    get: () => Promise<Response>;
+}
+
+export const MyServiceConfig: ContextDecorator<MyServiceConfig> = DI.createContext();
```

Comment:
I think the type annotation is unnecessary here. Passing the interface as a type parameter to createContext has the same result with less things needed to be imported.